util/certs/certs.go

```diff
@@ -76,11 +78,37 @@ func DecodeCertPEM(encoded []byte) (*x509.Certificate, error) {
 
 // DecodePrivateKeyPEM attempts to return a decoded key or nil
 // if the encoded input does not contain a private key.
-func DecodePrivateKeyPEM(encoded []byte) (*rsa.PrivateKey, error) {
+func DecodePrivateKeyPEM(encoded []byte) (crypto.Signer, error) {
```

Comment:
Should we add a test for validating the func can handle different types of private keys?

Reply:
I think that's just testing the compiler / the stdlib

Reply:
added, thank you